lightning-block-sync/src/http.rs

```diff
@@ -339,17 +334,26 @@ impl HttpClient {
 							reader.read_exact(&mut content[chunk_offset..]).await?;
 							content.resize(chunk_offset + chunk_size, 0);
 						}
-						Ok(content)
+						content
 					}
 					#[cfg(not(feature = "tokio"))]
 					{
 						let mut decoder = chunked_transfer::Decoder::new(reader);
 						decoder.read_to_end(&mut content)?;
-						Ok(content)
+						content
 					}
 				}
 			},
+		};
+
+		if !status.is_ok() {
+			// TODO: Handle 3xx redirection responses.
+			let error_msg = format!("Errored with status: {} and contents: {}",
+			                        status.code, String::from_utf8_lossy(&contents));
```

Comment:
I think we should first check that the bytes are in the printable ascii range (>= 0x20, <= 0x7e).

Reply:
Not sure I did this in the best way